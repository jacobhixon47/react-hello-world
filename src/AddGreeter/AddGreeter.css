.add-greeter {
  margin: 20px;
  padding: 20px;
  width: 100%;
  border: 2px solid #00FFD8;
  background: #DDFFEE;
  text-align: center;
}

input {
  padding: 1%;
  font-size: 1.5em;
}
