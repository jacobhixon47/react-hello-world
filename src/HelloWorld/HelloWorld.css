.hello-world {
  margin: 10px;
  padding-top: 20px;
  padding-bottom: 20px;
  width: 50%;
  flex-grow: 1;
  text-align: center;
  background-color: #223454;
  color: #dddddd;
}

.spanishify, .englishify, .frenchify, .remove {
  padding: 2%;
  border: none;
  width: 15%;
  margin: 5px;
  border-radius: 20px;
  color: #eeeeee;
}

.spanishify:hover, .englishify:hover, .frenchify:hover {
  background-color: #aaaaaa;
  cursor: pointer;
}

.spanishify {
  background-color: #553FFF;
}

.englishify {
  background-color: #009922;
}

.frenchify, .remove {
  background-color: #770F40;
}
