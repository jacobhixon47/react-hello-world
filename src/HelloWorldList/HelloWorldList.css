.hello-list {
  padding: 20px;
  margin: 30px;
  display: flex;
  align-items: center;
  justify-content: center;
  flex-wrap: wrap;
  border: 2px solid #00D8FF !important;
  background: #DDEEFF !important;
}
