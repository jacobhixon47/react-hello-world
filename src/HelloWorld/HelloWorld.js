import React, { Component } from 'react';
import './HelloWorld.css';

class HelloWorld extends Component {
  constructor(props) {
    super(props);
    this.state = {
      greeting: 'Hello'
    };
    this.frenchify = this.frenchify.bind(this);
    this.englishify = this.englishify.bind(this);
    this.spanishify = this.spanishify.bind(this);
    this.removeGreeting = this.removeGreeting.bind(this);
  }
  frenchify() {
    this.setState({ greeting: 'Bonjour' });
  }
  spanishify() {
    this.setState({ greeting: 'Hola' });
  }
  englishify() {
    this.setState({ greeting: 'Hello' });
  }
  removeGreeting() {
    this.props.removeGreeting(this.props.name);
  }
  render() {
    return (
      <div className="hello-world">
        {this.state.greeting}, {this.props.name}!
        <br/>
        <button className="frenchify" onClick={this.frenchify}>Frenchify!</button>
        <br/>
        <button className="spanishify" onClick={this.spanishify}>Spanishify!</button>
        <br/>
        <button className="englishify" onClick={this.englishify}>Englishify!</button>
        <br/>
        <button className="remove" onClick={this.removeGreeting}>Remove Me!</button>
      </div>
    );
  }
}

export default HelloWorld;
