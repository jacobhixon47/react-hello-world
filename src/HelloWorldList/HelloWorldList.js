import React, { Component } from 'react';
import HelloWorld from '../HelloWorld/HelloWorld';
import AddGreeter from '../AddGreeter/AddGreeter';
import './HelloWorldList.css'

class HelloWorldList extends Component {
  constructor(props) {
    super(props);
    this.state = { greetings: ['Bender'] };
    this.addGreeting = this.addGreeting.bind(this);
    this.removeGreeting = this.removeGreeting.bind(this);
  }
  addGreeting(newName) {
    this.setState({ greetings: [...this.state.greetings, newName] });
  }
  removeGreeting(removeName) {
    const filteredGreetings = this.state.greetings.filter(name => {
      return name !== removeName;
    });
    this.setState({ greetings: filteredGreetings });
  }
  renderGreetings() {
    return this.state.greetings.map(name => (
      <HelloWorld
        key={name}
        name={name}
        removeGreeting={this.removeGreeting}
      />
    ));
  }
  render() {
    return (
      <div className="hello-list">
        <AddGreeter addGreeting={this.addGreeting} />
        {this.renderGreetings()}
      </div>
    );
  }
}

export default HelloWorldList;
