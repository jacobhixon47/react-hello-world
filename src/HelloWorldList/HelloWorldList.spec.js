import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';

import HelloWorldList from './HelloWorldList';
import AddGreeter from '../AddGreeter/AddGreeter';
import HelloWorld from '../HelloWorld/HelloWorld';

describe(HelloWorldList, () => {
  const component = shallow(
    <HelloWorldList />
  );
// tests
  // snapshot test
  it('renders and matches snapshot', () => {
  // shallow render a component
    const component = renderer.create(
      <HelloWorldList />
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  // test that AddGreeter sub-component exists
  it('contains AddGreeter sub-component', () => {
    expect(component.find(AddGreeter)).toHaveLength(1);
  });
  // test for existence of a HelloWorld component for each greeting
  it('contains the same number of HelloWorld components as greetings', () => {
    const helloWorlds = component.find(HelloWorld).length;
    const greetings = component.state('greetings').length;
    expect(helloWorlds).toEqual(greetings);
  });
  // test addGreeting() functionality
  it('adds a greeting when addGreeting function is called', () => {
    const before = component.find(HelloWorld).length;
    component.instance().addGreeting('Sample');
    const after = component.find(HelloWorld).length;
    expect(after).toBeGreaterThan(before);
  });
  // test removeGreeting() functionality
  it('removes a greeting from the list when the remove greeting function is called', () => {
    const before = component.find(HelloWorld).length;
    const removeMe = component.state('greetings')[0];
    component.instance().removeGreeting(removeMe);
    const after = component.find(HelloWorld).length;
    expect(after).toBeLessThan(before);
  });
});
