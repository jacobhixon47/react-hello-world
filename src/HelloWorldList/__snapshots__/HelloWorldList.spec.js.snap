// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`class HelloWorldList extends _react.Component {
  constructor(props) {
    super(props);
    this.state = { greetings: ['Bender'] };
    this.addGreeting = this.addGreeting.bind(this);
    this.removeGreeting = this.removeGreeting.bind(this);
  }
  addGreeting(newName) {
    this.setState({ greetings: [...this.state.greetings, newName] });
  }
  removeGreeting(removeName) {
    const filteredGreetings = this.state.greetings.filter(name => {
      return name !== removeName;
    });
    this.setState({ greetings: filteredGreetings });
  }
  renderGreetings() {
    return this.state.greetings.map(name =>
    _react2.default.createElement(_HelloWorld2.default, {
      key: name,
      name: name,
      removeGreeting: this.removeGreeting, __source: { fileName: _jsxFileName, lineNumber: 24 }, __self: this }));


  }
  render() {
    return (
      _react2.default.createElement('div', { className: 'hello-list', __source: { fileName: _jsxFileName, lineNumber: 33 }, __self: this },
        _react2.default.createElement(_AddGreeter2.default, { addGreeting: this.addGreeting, __source: { fileName: _jsxFileName, lineNumber: 34 }, __self: this }),
        this.renderGreetings()));


  }} renders and matches snapshot 1`] = `
<div
  className="hello-list"
>
  <div
    className="add-greeter"
  >
    <input
      onChange={[Function]}
      type="text"
      value=""
    />
    <br />
    <br />
    <button
      onClick={[Function]}
    >
      Add
    </button>
  </div>
  <div
    className="hello-world"
  >
    Hello
    , 
    Bender
    !
    <br />
    <button
      className="frenchify"
      onClick={[Function]}
    >
      Frenchify!
    </button>
    <br />
    <button
      className="spanishify"
      onClick={[Function]}
    >
      Spanishify!
    </button>
    <br />
    <button
      className="englishify"
      onClick={[Function]}
    >
      Englishify!
    </button>
    <br />
    <button
      className="remove"
      onClick={[Function]}
    >
      Remove Me!
    </button>
  </div>
</div>
`;
